File: system/exec/bin/exec.cc
#include <iostream>
#include "../src/exec.h"
#include <string>
using namespace std;

int main () {
    printf("test");
    string s = exec("ls /bin");

    cout << s << endl;
    s = exec("echo 'aaa'");
    cout << s << endl; 

    s = exec("ls -la");
    cout << s << endl; 

    return 1;
}

File: tests/exec.js
var phplikeMod = require('./include.js');


var assert = require("assert");

//mocha lib/ --grep mthod_get
describe('Test function: exec', function() {
    it('Echo string', function() {
        var str = "";
        for (var i = 0; i < 10; i++) {
            str += "1";
        }
        var printInScreen = false;
        var res = phplikeMod.exec("echo " + str + "", printInScreen);
        assert.equal("1111111111\n", res);
    })

});

describe('Test function: system', function() {
    it('Echo string', function() {
        var str = "";
        for (var i = 0; i < 10; i++) {
            str += "1";
        }
        var printInScreen = false;
        var res = phplikeMod.system("echo " + str + "", printInScreen);
        assert.equal("1111111111\n", res);
    })

});




